### PR TITLE
FEATURE: filter admin reports via user group

### PR DESCRIPTION
**Pull request by techAPJ**
This PR adds group dropdown to filter Admin reports via user group. Currently these reports support group filter:

- User Visits
- New Users (Signups)
- User Profile Views

Screenshot:

screen shot 2016-02-04 at 01 58 07

Verified the filtered reports against Discourse Meta backup.

### CODE REVIEW

**Comment by techAPJ** [comment]
Build failure is unrelated.

File: app/assets/javascripts/admin/controllers/admin-reports.js.es6 [hunk 1]
```diff
@@ -16,15 +17,21 @@ export default Ember.Controller.extend({
     return arr.concat( Discourse.Site.currentProp('sortedCategories').map(function(i) { return {name: i.get('name'), value: i.get('id') }; }) );
   }.property(),
 
+  groupOptions: function() {
+    var arr = [{name: I18n.t('admin.dashboard.reports.groups'), value: 'all'}];
+    return arr.concat( this.site.groups.map(function(i) { return {name: i['name'], value: i['id'] }; }) );
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-reports.js.es6]: Nitpick here. The brackets and parentheses are not consistently spaced. Also...`ES6` functions` \o/ (i) =>`

File: app/assets/javascripts/admin/controllers/admin-reports.js.es6 [hunk 2]
```diff
@@ -16,15 +17,21 @@ export default Ember.Controller.extend({
     return arr.concat( Discourse.Site.currentProp('sortedCategories').map(function(i) { return {name: i.get('name'), value: i.get('id') }; }) );
   }.property(),
 
+  groupOptions: function() {
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-reports.js.es6]:
```
groupOptions() {
  //
}
```

File: app/assets/javascripts/admin/controllers/admin-reports.js.es6 [hunk 3]
```diff
@@ -16,15 +17,21 @@ export default Ember.Controller.extend({
...
+    var arr = [{name: I18n.t('admin.dashboard.reports.groups'), value: 'all'}];
+    return arr.concat( this.site.groups.map(function(i) { return {name: i['name'], value: i['id'] }; }) );
+  }.property(),
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-reports.js.es6]: We can `import computed from 'ember-addons/ember-computed-decorators';`

File: app/assets/javascripts/admin/controllers/admin-reports.js.es6 [hunk 4]
```diff
@@ -16,15 +17,21 @@ export default Ember.Controller.extend({
...
+
+  showGroupOptions:function() {
+    return this.get("model.type") === "visits" || this.get("model.type") === "signups" || this.get("model.type") === "profile_views";
+  }.property("model.type"),
```
Comment (tgxworld) [inline app/assets/javascripts/admin/controllers/admin-reports.js.es6]: one nice thing about about ember-computed-decorators is that you can do

```
@computed('model.type')
showGroupOptions(modelType) {
 // No need for this.get("modelType")
}
```

**Comment by techAPJ** [comment]
@tgxworld thanks for the tips :+1: PR updated

**Pull request merged**